### PR TITLE
Fix several deprecation warnings and add samples to solution file

### PR DESCRIPTION
**Pull request by msoeken**
This PR fixes deprecation warnings in:

* `ReversibleLogicSynthesis`
* `OrderFinding`
* `OracleSynthesis`
* `IntegerFactorization`

One deprecation warning remains in `IntegerFactorization` due to a use of a deprecated function in Libraries. Otherwise, I couldn't spot any other occurrence of a deprecated function use.

It also adds `OrderFinding` and `OracleSynthesis` to the solution file.

The PR further contains some small code changes to make it more functional.

### CODE REVIEW

File: Samples/src/OrderFinding/OrderFinding.qs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License.
 
 namespace Microsoft.Quantum.Samples.OrderFinding {
+	open Microsoft.Quantum.Arrays;
```
Comment (cgranade) [inline Samples/src/OrderFinding/OrderFinding.qs]: It looks like there's a slight tabs vs spaces issue here...

File: Samples/src/OrderFinding/OrderFinding.qs [hunk 2]
```diff
@@ -25,11 +26,7 @@ namespace Microsoft.Quantum.Samples.OrderFinding {
     /// Square([2, 3, 0, 1]), // [0, 1, 2, 3]
     /// ```
     function Square(perm : Int[]) : Int[] {
-        mutable squaredPermutation = new Int[Length(perm)];
-        for (i in 0..Length(perm) - 1) {
-            set squaredPermutation w/= i <- perm[perm[i]];
-        }
-        return squaredPermutation;
+		return Mapped(LookupFunction(perm), perm);
```
Comment (cgranade) [inline Samples/src/OrderFinding/OrderFinding.qs]: More tab/spaces here.

File: Samples/src/OrderFinding/OrderFinding.qs (hunk 1 above)
Comment (msoeken) [reply Samples/src/OrderFinding/OrderFinding.qs]: I tried a new IDE and didn't spot these. Fixed them now and checked all other changed files as well.

File: Samples/src/OrderFinding/OrderFinding.qs (hunk 2 above)
Comment (msoeken) [reply Samples/src/OrderFinding/OrderFinding.qs]: Fixed them in 157ded4

**Pull request merged**